lib/test.js

```diff
@@ -53,6 +53,77 @@ function plan(count) {
 	this.plan(count, captureStack(this.plan));
 }
 
+function tryTest(fn) {
+	const args = [];
+	for (let i = 1, len = arguments.length; i < len; i++) {
+		args.push(arguments[i]);
+	}
+
+	let commitCalled = false;
+	let discarded = false;
+	let finished = false;
+
+	const commitFn = (ret, toCommit) => {
+		if (commitCalled) {
+			if (discarded) {
+				this.saveFirstError(new Error('The discard() was already called'));
+			} else {
+				this.saveFirstError(new Error('The commit() was already called'));
+			}
+			return;
+		}
+		commitCalled = true;
+
+		if (ret.passed) {
+			this.countFinishedAttemptAssertion(toCommit);
+		} else {
+			this.countFailedAttemptAssertion(ret.error, toCommit);
+		}
+
+		if (toCommit) {
+			for (const log of ret.logs) {
+				this.addLog(log);
+			}
+		}
+	};
+
+	const running = this
+		.forAttempt(t => fn.apply(null, [t].concat(args)))
+		.run();
+
+	this.addPendingAttemptAssertion();
+
+	const finishedRunning = running.then(ret => {
+		if (discarded) {
+			return null;
+		}
+
+		finished = true;
+		return Object.assign({}, ret, {
+			commit: () => commitFn(ret, true),
```

Comment:
This seems to be missing a few properties that are declared in the TypeScript definition.

Reply:
they come from the `ret` value.

Reply:
Yes of course.